ui-core/src/components/EditAddress.tsx

```diff
@@ -16,13 +16,15 @@ export function EditAddress(
     setMailingAddress,
     validationResult,
     showLabels = true,
-    language = 'en' // todo: accept internationalized labels, see JN-910 for implementation
+    language = 'en',
+    i18n
   }: {
     mailingAddress: MailingAddress,
     setMailingAddress: (updated: React.SetStateAction<MailingAddress>) => void,
     validationResult?: AddressValidationResult,
     showLabels: boolean,
-    language: string
+    language: string,
+    i18n?: (key: string) => string
```

Comment:
the idea of making i18n as a context is that we don't have to pass it as an argument.  So this should just be `const { i18n } = useI18n()` below

Reply:
This is in core, though - so it can't do a `useI18n`. Happy to see if there's some way to refactor it, but I am not sure there is cause it needs to hit the participant API.

Reply:
ah, thanks, I hadn't noticed that.    Let's refactor I18NProvider to take an ApiProvider (see `ui-core/src/participant/ApiProvider.tsx`) and extend ApiProvider to have a `getLanguageTexts` method.  this will both help clean up this code, and make it easier to enhance the admin-side previews of i18n functions

Reply:
So - `i18nProvider` would be in core? Just to make sure I understand. And in the admin API, should there be a separate endpoint for getting language texts like the participant one?

Reply:
Yeah, I think that's right.

Reply:
@connorlbark to save you the hassle, here's a branch with this done: https://github.com/broadinstitute/juniper/pull/792

Reply:
@MatthewBemis this is awesome! should we make a ticket and merge it first?

Reply:
done, that PR has a ticket and is ready for review now